### PR TITLE
Add wgRestrictionLevels to ManageWiki per T5991

### PR DESCRIPTION
**Pull request by Universal-Omega**
I believe this will work, but not 100% certain.

### CODE REVIEW

**Review by RhinosF1 — COMMENTED** [review]
What if wikis want more custom named ones?

**Comment by Universal-Omega** [comment]
It should work with custom ones also per `'type' => 'usergroups'`

**Comment by Universal-Omega** [comment]
Note: I did not remove the existing config for this that wikis did through GitHub PRs - I was unsure if that was needed it not, but in the past when moving configs to ManageWiki, they were removed, so I thought I'd mention that.

**Comment by Reception123** [comment]
I tested this on test2 by adding the bureaucrat group as a restriction level and it didn't seem to work

**Comment by Reception123** [comment]
My guess would be because https://github.com/miraheze/mw-config/blob/master/LocalSettings.php#L2262 is not fulfilled (it seems that for restriction levels the group must have its own name as a "user right").

File: ManageWikiSettings.php [hunk 1]
```diff
@@ -349,6 +349,23 @@
 		'section' => 'edit',
 		'help' => 'Show more authors.',
 	],
+	'wgRestrictionLevels' => [
+		'name' => 'Restriction Levels',
+		'from' => 'mediawiki',
+		'restricted' => false,
+		'type' => 'userrights',
+		'options' => [
+			'no protection' => '',
```
Comment (paladox) [inline ManageWikiSettings.php:358]: Need to provide options here. The below part only makes sure it's the default.

File: ManageWikiSettings.php [hunk 2]
```diff
@@ -349,6 +349,23 @@
 		'section' => 'edit',
 		'help' => 'Show more authors.',
 	],
+	'wgRestrictionLevels' => [
+		'name' => 'Restriction Levels',
+		'from' => 'mediawiki',
+		'restricted' => false,
+		'type' => 'userrights',
```
Comment (paladox) [inline ManageWikiSettings.php:356]: should be list-multi

File: ManageWikiSettings.php (hunk 1 above)
Comment (paladox) [reply ManageWikiSettings.php:358]: Add all the options that wikis are using see https://github.com/miraheze/mw-config/blob/1e327aade41e4a316a6fda8394e243696e232581/LocalSettings.php#L2867

**Pull request closed without merging**